test/test_models.py

```diff
@@ -69,16 +91,255 @@ def check_script(self, model, name):
+        test_names = self.get_test_methods_for_class(ClassificationModelTester)
+
+        for model_name in model_names:
+            test_name = 'test_' + model_name
+            self.assertTrue(test_name in test_names)
+
+
+class ClassificationModelTester(ModelTester):
+    def _infer_for_test_with(self, model, test_input):
+        return model(test_input)
+
+    def _check_classification_output_shape(self, test_output, num_classes):
+        self.assertEqual(test_output.shape, (1, num_classes))
+
+    # NOTE Depends on presence of test data fixture. See common_utils.py for
+    #    details on creating fixtures.
+    def _check_model_correctness(self, model, test_input, num_classes=STANDARD_NUM_CLASSES):
+        test_output = self._infer_for_test_with(model, test_input)
+        self._check_classification_output_shape(test_output, num_classes)
+        self.assertExpected(test_output)
+        return test_output
+
+    # NOTE override this in a child class
+    def _get_input_shape(self):
+        return STANDARD_INPUT_SHAPE
+
+    def _test_classification_model(self, model_callable, num_classes=STANDARD_NUM_CLASSES, **kwargs):
+        model = self._get_test_model(model_callable, num_classes=num_classes, **kwargs)
+        self._check_scriptable(model, True)  # currently, all expected to be scriptable
+        test_input = self._get_test_input(shape=self._get_input_shape())
+        test_output = self._check_model_correctness(model, test_input)
+        return model, test_input, test_output
+
+
+class AlexnetTester(ClassificationModelTester):
+    def test_alexnet(self):
+        self._test_classification_model(models.alexnet)
+
+
+# TODO add test for aux_logits arg to factory method
+# TODO add test for transform_input arg to factory method
+class InceptionV3Tester(ClassificationModelTester):
+    def _get_input_shape(self):
+        return (1, 3, 299, 299)
+
+    def test_inception_v3(self):
+        self._test_classification_model(models.inception_v3)
+
+
+class SqueezenetTester(ClassificationModelTester):
+    def test_squeezenet1_0(self):
+        self._test_classification_model(models.squeezenet1_0)
+
+    def test_squeezenet1_1(self):
+        self._test_classification_model(models.squeezenet1_1)
+
+
+# TODO add test for width_mult arg to factory method
+class MobilenetTester(ClassificationModelTester):
+    def test_mobilenet_v2(self):
+        self._test_classification_model(models.mobilenet_v2)
+
+    def test_mobilenetv2_residual_setting(self):
+        self._test_classification_model(models.mobilenet_v2, inverted_residual_setting=[[1, 16, 1, 1], [6, 24, 2, 2]])
+
+
+# TODO add test for aux_logits arg to factory method
+# TODO add test for transform_input arg to factory method
+class GooglenetTester(ClassificationModelTester):
+    def test_googlenet(self):
+        self._test_classification_model(models.googlenet)
+
+
+class VGGNetTester(ClassificationModelTester):
+    def test_vgg11(self):
```

Comment:
I'm not sure this is an improvement.

Reply:
See my long note below for why I believe it is.